domain/secret/state/state.go

```diff
@@ -635,22 +634,47 @@ UNION
     SELECT MAX(revision) FROM secret_revision rev
     WHERE  rev.secret_id = $secretRef.secret_id
 )
-UPDATE secret_revision
-SET    obsolete = true,
-       pending_delete = true,
-       update_time = DATETIME('now')
-WHERE  secret_id = $secretRef.secret_id
-AND    obsolete = false
-AND    revision NOT IN (SELECT * FROM in_use)`, secretRef{})
+SELECT sr.uuid AS &secretRevision.uuid
+FROM   secret_revision sr
+WHERE  sr.secret_id = $secretRef.secret_id
+AND    sr.revision NOT IN (SELECT * FROM in_use)`, secretRef{}, secretRevision{})
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	stmt, err := st.Prepare(`
+INSERT INTO secret_revision_obsolete (*)
+VALUES ($secretRevisionObsolete.*)
+ON CONFLICT(revision_uuid) DO UPDATE SET
+    obsolete=excluded.obsolete,
+    pending_delete=excluded.pending_delete`, secretRevisionObsolete{})
 	if err != nil {
 		return errors.Trace(err)
 	}
-	err = tx.Query(ctx, stmt, secretRef{ID: uri.ID}).Run()
+
+	var revisionUUIIDs secretRevisions
+	err = tx.Query(ctx, query, secretRef{ID: uri.ID}).GetAll(&revisionUUIIDs)
+	st.logger.Warningf("markObsoleteRevisions: %v, err %#v", revisionUUIIDs, err)
 	if errors.Is(err, sqlair.ErrNoRows) {
 		// No obsolete revisions to mark.
 		return nil
 	}
-	return errors.Trace(err)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	for _, revisionUUID := range revisionUUIIDs {
```

Comment:
sqlair now supports bulk insert. can you add a todo to use it or make the change in this PR